### PR TITLE
HM-52 [Auth] Make Preferences Entity

### PR DESCRIPTION
**Pull request by ChrisIvanov**

### CODE REVIEW

File: src/Auth/Auth.Infrastucture/Persistence/ApplicationDbContext.cs [hunk 1]
```diff
@@ -54,6 +54,37 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                     .HasColumnName("user_id")
                     .IsRequired(true);
             });
+
+            modelBuilder.Entity<Preferences>(entity =>
+            {
+                entity.ToTable("Preferences", "security");
+                entity.HasKey(pref => pref.Id);
+
```
Comment (rosteslav) [inline src/Auth/Auth.Infrastucture/Persistence/ApplicationDbContext.cs:85]: describe id as well
`entity.Property(e => e.Id)`
`    .HasColumnName("id");`

File: src/Auth/Auth.Domain/Entities/Preferences.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿namespace BuildingMarket.Auth.Domain.Entities
+{
+    public class Preferences
+    {
+        public int Id { get; set; }
+        public string UserId { get; set; }
+        public string Purpose { get; set; }
+        public string Region { get; set; }
+        public string BuildingType { get; set; }
+        public string PriceHigherEnd { get; set; }
```
Comment (rosteslav) [inline src/Auth/Auth.Domain/Entities/Preferences.cs]: decimal

File: src/Auth/Auth.Infrastucture/Persistence/ApplicationDbContext.cs [hunk 3]
```diff
@@ -54,6 +54,37 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+                .HasMaxLength(255);
+
+                entity.Property(pref => pref.BuildingType)
+                .HasColumnName("building_type")
+                .HasColumnType("character varying")
+                .HasMaxLength(255);
+
+                entity.Property(pref => pref.PriceHigherEnd)
+                .HasColumnName("price_higher_end")
+                .HasColumnType("character varying")
```
Comment (rosteslav) [inline src/Auth/Auth.Infrastucture/Persistence/ApplicationDbContext.cs]: real .. I think

File: src/Auth/Auth.Domain/Entities/Preferences.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿namespace BuildingMarket.Auth.Domain.Entities
+{
+    public class Preferences
+    {
+        public int Id { get; set; }
+        public string UserId { get; set; }
```
Comment (rosteslav) [inline src/Auth/Auth.Domain/Entities/Preferences.cs]: required, the C# keyword

File: src/Auth/Auth.Domain/Entities/Preferences.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿namespace BuildingMarket.Auth.Domain.Entities
+{
+    public class Preferences
+    {
+        public int Id { get; set; }
```
Comment (rosteslav) [inline src/Auth/Auth.Domain/Entities/Preferences.cs]: reuquired

**Review by rosteslav — CHANGES_REQUESTED** [review]

File: src/Auth/Auth.Infrastucture/Persistence/ApplicationDbContext.cs (hunk 1 above)
Comment (ChrisIvanov) [reply src/Auth/Auth.Infrastucture/Persistence/ApplicationDbContext.cs:85]: Should I state a type for the Id column?

**Review by rosteslav — APPROVED** [review]

**Pull request merged**